polar-core/src/vm.rs

```diff
@@ -2007,14 +2059,16 @@ impl PolarVirtualMachine {
     /// This is sort of a "sub-goal" of `Unify`.
     fn unify_partial(&mut self, partial: &partial::Constraints, right: &Term) -> PolarResult<()> {
         let mut partial = partial.clone();
-        if let Value::Partial(right_partial) = right.value() {
```

Comment:
Was this code wrong?

Reply:
I think so. I think that if you unify two partials, we should make their constraint sets identical thereafter, either by sharing a ref, or adding constraints to the other whenever we add them to one, or something. That seems medium hard, so we just made it an error for now.